core/backend/src/test/standalone/Settings.test.ts

```diff
@@ -183,4 +184,39 @@ describe("Settings", () => {
     expect(appSettings.getArray<string>("cSpell/enableFiletypes")!.length).equals(17);
     appSettings.dropDictionary(settingFileName);
   });
+
+  it("IModel persistent settings ", () => {
+    const iModelName = IModelTestUtils.prepareOutputFile("IModel", "test.bim");
+    const iModel2 = IModelTestUtils.createSnapshotFromSeed(iModelName, IModelTestUtils.resolveAssetFile("test.bim"));
+
+    const setting2: SettingDictionary = {
+      "workspace/container/alias": [
+        { name: "default-icons", id: "icons-01" },
+        { name: "default-lang", id: "lang-05" },
+        { name: "default-fonts", id: "fonts-02" }, // a container id that doesn't exist
+        { name: "default-key", id: "key-05" },
+      ],
+    };
+    iModel2.saveSettingDictionary("testSetting", setting2);
+    iModel2.saveSettingDictionary("test2", imodel1Settings);
+    iModel2.close();
+
+    let iModel3 = StandaloneDb.openFile(iModelName, OpenMode.ReadWrite);
+    expect(iModel3.workspace.settings.getObject("workspace/container/alias")).to.deep.equal(setting2["workspace/container/alias"]);
+    expect(iModel3.workspace.settings.getString("app1/sub1")).equal("imodel1 value");
+
+    iModel3.saveSettingDictionary("test2", imodel2Settings);
+    iModel3.close();
+    iModel3 = StandaloneDb.openFile(iModelName);
+    expect(iModel3.workspace.settings.getObject("workspace/container/alias")).to.deep.equal(setting2["workspace/container/alias"]);
+    expect(iModel3.workspace.settings.getString("app1/sub1")).equal("imodel2 value");
+    iModel3.deleteSettingDictionary("test2");
+    iModel3.close();
+
+    iModel3 = StandaloneDb.openFile(iModelName);
+    expect(iModel3.workspace.settings.getObject("workspace/container/alias")).to.deep.equal(setting2["workspace/container/alias"]);
+    expect(iModel3.workspace.settings.getString("app1/sub1")).to.be.undefined;
```

Comment:
```
  1) Settings
       IModel persistent settings :
     AssertionError: expected 'app1 value' to be undefined
      at Context.<anonymous> (src\test\standalone\Settings.test.ts:218:68)
      at processImmediate (internal/timers.js:464:21)
```

Reply:
fixed. Can you pull my changes and see if you can reproduce the assert?